plugin/src/main/java/com/denizenscript/denizen/events/entity/AreaEffectCloudApplyScriptEvent.java

```diff
@@ -0,0 +1,63 @@
+package com.denizenscript.denizen.events.entity;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.EntityTag;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ListTag;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.AreaEffectCloudApplyEvent;
+
+public class AreaEffectCloudApplyScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // area effect cloud applies
+    //
+    // @Group Entity
+    //
+    // @Cancellable true
+    //
+    // @Triggers when an area_effect_cloud tries to apply its effect(s) to entities within range.
+    //
+    // @Warning
+    // This runs every 5 ticks if there are any entities in the area effect cloud's bounding box. Prefer <@link event entity potion effects modified> for listening to normal potion effect changes.
+    //
+    // @Context
+    // <context.entity> returns the EntityTag of the area effect cloud.
+    // <context.affected_entities> returns a ListTag of EntityTags affected by the area effect cloud. Note that this can be empty, and only lists which entities are currently being refreshed.
+    //
+    // @Determine
+    // "AFFECTED_ENTITIES:<ListTag(EntityTag)>" to determine the entities that will be affected by the area effect cloud.
+    //
+    // -->
+
+    public AreaEffectCloudApplyScriptEvent() {
+        registerCouldMatcher("area effect cloud applies");
+        this.<AreaEffectCloudApplyScriptEvent, ListTag>registerDetermination("affected_entities", ListTag.class, (evt, context, list) -> {
+            evt.event.getAffectedEntities().clear();
+            for (EntityTag entity : list.filter(EntityTag.class, context)) {
+                evt.event.getAffectedEntities().add(entity.getLivingEntity());

```

Comment:
This only allows living entities, but doesn't have any proper check/error, I.e. specifying the wrong entity type will just error at some point along the way (either due to a `null` value or a cast exception) - should have a proper check in here and ideally mention that in the meta as well.